operator/pkg/translate/translate.go

```diff
@@ -609,6 +609,10 @@ func (t *Translator) TranslateHelmValues(iop *v1alpha1.IstioOperatorSpec, compon
 		for k, v := range components {
 			finalVals[k] = v
 		}
+		// this seems hacky to me but if we want to have even basic support for revisions
+		//    I think we need to do something to pass the value down to components which use
+		//    FlattenValues
+		finalVals["revision"] = mergedVals["revision"]
```

Comment:
tbh I don't understand the code but isn't https://github.com/istio/istio/blob/bd9e52d05817b03b9811a761a818b2e76d356018/operator/pkg/translate/translate.go#L117 supposed to handle it?

Reply:
Unless I'm missing something overwriting `mergedVals` on 616 effectively tosses all the previous work and only includes the finalVals we've created in this block which are under global and ztunnel (or whatever the current component being flattened is). Since revisions is top level it's not included and needs to be added back.

In practice until I added this line I was not able to use `.Values.revision` in the go/helm template so I think _something_ is needed to add that back in or prevent stripping it out. Overall the flattening is perhaps being more aggressive than we intended and maybe a better solution revolves around adjusting that rather than explicitly plucking single desired fields ad-hoc.

Reply:
And to be clear I'm happy to work on another solution to the flattening stuff that's cleaner if the overall concept of adjusting our installs for ztunnel to support revisions is generally well accepted. If the feedback on the concept was very negative I didn't want to have spent too much time so it's shown here with a bit of a workaround.

Reply:
I think the concept is fine, want to let others take a look though

Reply:
Sure. I'll let this sit for a tick and collect feedback before I invest more time in refining the golang. Thanks for taking a look!

Reply:
its hard to move the others due to compatibiltiy issues. But the unflattened is basically blatantly incorrect Helm practice

Reply:
@howardjohn, I took a crack at a solution which fits more closely with the rest of the flattening logic. LMK what you think when you have a chance.